src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -746,13 +746,22 @@ public function offsetGet($option)
             }
 
             if (!$valid) {
-                $keys = array_keys($invalidTypes);
+                $fmtActualValue = $this->formatValue($value);
+                $fmtAllowedTypes = implode('" or "', $allowedTypes);
+                $fmtProvidedTypes = implode('|', array_keys($invalidTypes));
+
+                $allowedContainsArrayType = \count(array_filter(
```

Comment:
Is this logic needed? Wouldn't something like `$allowedContainsArrayType = '[]' === substr($type, -2);` be enough?

Reply:
Actually, it's not enough, see test `testResolveFailsIfInvalidType` and data set in `575` and `576` lines.

Reply:
@xabbuh ping